### PR TITLE
feat(www): updated landing page

### LINKED ISSUE #67: Landing page redesign
**Is your feature request related to a problem? Please describe.**
Current landing page lacks a lot of character and has a lot of whitespace.

**Describe the solution you'd like**
Redesign.

**Describe alternatives you've considered**
Keep as is.

### PR DESCRIPTION
**Pull request by crherman7**
## Scope

[closes TICKET-67](https://github.com/crherman7/rechunk/issues/67)

## Implementation

This pull request introduces significant improvements to the UI components and landing page of the ReChunk website. The following changes have been made:

**1. New Components**
- **`Header`**: A sticky, responsive header with navigation links to GitHub, Discord, and the project documentation.
- **`GridPattern`**: A reusable SVG-based grid component for background patterns with customizable properties.
- **`Iphone15Pro`**: A scalable SVG component to visually represent an iPhone mockup for UI previews.
- **`Safari`**: A Safari browser-like SVG component to display content previews in a browser frame.
- **`ShinyDiv`**: An animated component using Framer Motion, designed to add visual appeal to buttons or sections.

**2. Landing Page Enhancements**
- Improved animations with `Framer Motion` for better engagement.
- Added `ShinyDiv` elements to highlight key sections of the page.
- Introduced tooltips to provide contextual hints for call-to-action buttons.
- Reorganized the layout for clarity and modularity, separating UI logic into reusable components.

**3. Updates to Existing Components**
- Updated the `text` component to increase `H1` font size, enhancing readability across devices.

**4. Asset Management**
- Added a new app screenshot (`screenshot.png`) for showcasing the functionality of the app.
- Removed outdated assets (`feed.png`, `iphone.png`) to declutter the project and focus on modern visuals.

**5. Code Improvements**
- Refactored the landing page (`routes/_index.tsx`) to utilize new components and improve maintainability.
- Replaced static HTML structures with reusable components for better code organization and reuse.

**Summary**
These updates collectively enhance the user experience, improve visual appeal, and make the codebase more modular and maintainable. The new components and redesign aim to better communicate the value of ReChunk to developers and stakeholders while providing a cleaner and more engaging UI.

## Screenshots

<details>
  <summary>desktop</summary>

  

Screenshot 2024-11-19 at 7 14 36 PM
</details>

<details>
  <summary>mobile</summary>
  
  

Screen Shot 2024-11-19 at 19 15 04
</details>

## How to Test

- `pnpm install`
- `cd apps/www`
- `cp .env.example .env`
- `pnpm run setup`
- `pnpm dev`
- Verify app renders in a browser

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

### CODE REVIEW

File: apps/www/app/routes/_index.tsx [hunk 1]
```diff
@@ -18,108 +25,129 @@ export const meta: MetaFunction = () => {
...
-      </footer>
     </div>
   );
 }
+
+/**
+ * ButtonLink Component
+ * Reusable component for navigation buttons.
+ */
+function ButtonLink({
```
Comment (devs9) [inline apps/www/app/routes/_index.tsx]: 🟡 I see some warnings for this button in the console, can we fix it?
and maybe it should be moved to `ui` if it's the reusable component?

File: apps/www/app/routes/_index.tsx [hunk 2]
```diff
@@ -18,108 +25,129 @@ export const meta: MetaFunction = () => {
...
-          <img
-            src="/arrow_11.svg"
-            className="size-20 rotate-[270deg] fixed mt-[340px] ml-28"
-            alt="arrow"
+        </ShinyDiv>
+        <Text.H1 id="myElement" className="text-center opacity-0">
+          Launch without limits.
+        </Text.H1>
+        <Text.P
+          id="myElement"
```
Comment (devs9) [inline apps/www/app/routes/_index.tsx]: 🤔 what do you think about renaming this?

File: apps/www/app/routes/_index.tsx (hunk 1 above)
Comment (crherman7) [reply apps/www/app/routes/_index.tsx]: you right!

File: apps/www/app/routes/_index.tsx (hunk 2 above)
Comment (crherman7) [reply apps/www/app/routes/_index.tsx]: good call!

File: apps/www/app/routes/_index.tsx (hunk 1 above)
Comment (crherman7) [reply apps/www/app/routes/_index.tsx]: updated :)

File: apps/www/app/routes/_index.tsx (hunk 2 above)
Comment (crherman7) [reply apps/www/app/routes/_index.tsx]: updated :)

File: apps/www/app/routes/_index.tsx (hunk 1 above)
Comment (crherman7) [reply apps/www/app/routes/_index.tsx]: The issue here is related to a `<button />` being a child of a `<button />` which can be seen from the radix component: https://discord.com/channels/@me/1242614275894153228/1309550040087531561. Likely would need to use `asChild` on the `<Tooltip.Trigger asChild>`.

This is fixed in the auth PR as tooltip is removed from the button.

Comment (crherman7) [reply apps/www/app/routes/_index.tsx]: great find on this error as well!

**Pull request merged**